internal/weapons/catalyst/flowingpurity/flowingpurity.go

```diff
@@ -0,0 +1,98 @@
+package flowingpurity
+
+import (
+	"fmt"
+
+	"github.com/genshinsim/gcsim/pkg/core"
+	"github.com/genshinsim/gcsim/pkg/core/attributes"
+	"github.com/genshinsim/gcsim/pkg/core/event"
+	"github.com/genshinsim/gcsim/pkg/core/keys"
+	"github.com/genshinsim/gcsim/pkg/core/player/character"
+	"github.com/genshinsim/gcsim/pkg/core/player/weapon"
+	"github.com/genshinsim/gcsim/pkg/modifier"
+)
+
+func init() {
+	core.RegisterWeaponFunc(keys.FlowingPurity, NewWeapon)
+}
+
+type Weapon struct {
+	Index int
+}
+
+func (w *Weapon) SetIndex(idx int) { w.Index = idx }
+func (w *Weapon) Init() error      { return nil }
+
+// When using an Elemental Skill, All Elemental DMG Bonus will be increased by 8/10/12/14/16% for 15s,
+// and a Bond of Life worth 24% of Max HP will be granted. This effect can be triggered once every 10s.
+// When the Bond of Life is cleared, every 1,000 HP cleared in the process will provide 2/2.5/3/3.5/4% All Elemental DMG Bonus,
+// up to a maximum of 12/15/18/21/24%. This effect lasts 15s.
+
+func NewWeapon(c *core.Core, char *character.CharWrapper, p weapon.WeaponProfile) (weapon.Weapon, error) {
+	w := &Weapon{}
+	r := p.Refine
+
+	const icdKey = "flowingpurity-icd"
+	const bondKey = "flowingpurity-bond"
+	eledmg := 0.06 + float64(r)*0.02
+	duration := 15 * 60
+	icd := 10 * 60
+
+	m := make([]float64, attributes.EndStatType)
+	bond := make([]float64, attributes.EndStatType)
+	hp := 0.24 //hpdebt_percentage
+	bondPercentage := 0.015 + float64(r)*0.005
+	bondDMGPCap := 0.09 + float64(r)*0.03
+
+	char.SetHPDebtByRatio(hp)
+	debt := char.CurrentHPDebt()
+	bondDMGP := (debt / 1000) * bondPercentage //use hp debt since you only get the buff after clearing bond anyway
+	if bondDMGP > bondDMGPCap {
+		bondDMGP = bondDMGPCap
+	}
```

Comment:
This needs to be moved inside of the OnSkill event subscription, because otherwise the bond will be granted on sim start and never again.